src/it/scala/com/wavesplatform/it/BalancesSpecification.scala

```diff
@@ -0,0 +1,299 @@
+
+      _ <- assertBalances(firstAddress, 190 waves, 90 waves)
+      _ <- assertBalances(secondAddress, 100 waves, 200 waves)
+
+      transferFailureAssertion <- recoverToSucceededIf[IOException] {
+        sender.transfer(firstAddress, secondAddress, 91 waves, fee = 1 waves)
+      }
+
+      _ <- Future.traverse(allNodes)(_.waitForHeight(fb + 2))
+
+      _ <- assertBalances(firstAddress, 190 waves, 90 waves)
+      _ <- assertBalances(secondAddress, 100 waves, 200 waves)
+    } yield transferFailureAssertion
+
+    Await.result(f, 1 minute)
+  }
+
+  test("can not make leasing without having enough waves for fee") {
+    val f = for {
+      fb <- Future.traverse(allNodes)(_.height).map(_.min)
+
+      _ <- assertBalances(firstAddress, 190 waves, 90 waves)
+      _ <- assertBalances(secondAddress, 100 waves, 200 waves)
+
+      transferFailureAssertion <- recoverToSucceededIf[IOException] {
+        sender.lease(firstAddress, secondAddress, 88 waves, fee = 3 waves)
+      }
+
+      _ <- Future.traverse(allNodes)(_.waitForHeight(fb + 2))
+
+      _ <- assertBalances(firstAddress, 190 waves, 90 waves)
+      _ <- assertBalances(secondAddress, 100 waves, 200 waves)
+    } yield transferFailureAssertion
+
+    Await.result(f, 1 minute)
+  }
+
+
+  test("leasing cancel is working correctly") {
+    val f = for {
+      _ <- assertBalances(firstAddress, 190 waves, 90 waves)
+      _ <- assertBalances(secondAddress, 100 waves, 200 waves)
+
+      createdLeaseTxId <- sender.lease(firstAddress, secondAddress, 70 waves, fee = 10 waves).map(_.id)
+
+      _ <- Future.traverse(allNodes)(_.waitForTransaction(createdLeaseTxId))
+
+      _ <- assertBalances(firstAddress, 180 waves, 10 waves)
+      _ <- assertBalances(secondAddress, 100 waves, 270 waves)
+
+      createdCancelLeaseTxId <- sender.cancelLease(firstAddress, createdLeaseTxId, fee = 10 waves).map(_.id)
+
+      _ <- Future.traverse(allNodes)(_.waitForTransaction(createdCancelLeaseTxId))
+
+      _ <- assertBalances(firstAddress, 170 waves, 70 waves)
+      _ <- assertBalances(secondAddress, 100 waves, 200 waves)
+    } yield succeed
+
+    Await.result(f, 1 minute)
+  }
+
+  test("assets issue should lead to a change in the asset balance and to a change of the waves balance and the effective balance only on the fee") {
+    val f = for {
+      _ <- assertBalances(firstAddress, 170 waves, 70 waves)
+
+      issuedAssetId <- sender.issue(firstAddress, "name", "description", 100000, 2, reissuable = true, fee = 10 waves).map(_.id)
+
+      _ <- Future.traverse(allNodes)(_.waitForTransaction(issuedAssetId))
+
+      _ <- assertBalances(firstAddress, 160 waves, 60 waves)
+      _ <- assertAssetBalance(firstAddress, issuedAssetId, 100000)
+    } yield succeed
+
+    Await.result(f, 1 minute)
+  }
+
+  test("assets reissue should lead to a change in the asset balance and a change of the waves balance and the effective balance only on the fee") {
```

Comment:
"asset reissue changes issuer's asset balance; issuer's waves balance is decreased by fee"